libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -0,0 +1,269 @@
+  { language = LanguageCode.ENGLISH, content = /.*/ }: LanguageStringQueryParams
+): ResultsReporting.LanguageString | undefined {
+  function textContentFilter(entry: ResultsReporting.LanguageString): boolean {
+    return !!entry.Content.match(content);
+  }
+  function languageFilter(entry: ResultsReporting.LanguageString): boolean {
+    return entry.Language === language;
+  }
+
+  return textEntries.find(
+    (entry: ResultsReporting.LanguageString) =>
+      textContentFilter(entry) && languageFilter(entry)
+  );
+}
+
+export function findBallotMeasureSelectionWithContent(
+  content: RegExp,
+  ballotMeasureSelections: readonly ResultsReporting.BallotMeasureSelection[]
+): ResultsReporting.BallotMeasureSelection {
+  return assertDefined(
+    ballotMeasureSelections.find((selection) => {
+      const internationalizedText = selection.Selection;
+      return !!findLanguageString(internationalizedText.Text, {
+        content,
+      });
+    }),
+    `Could not find ballot measure selection with content "${content.toString()}"`
+  );
+}
+
+function findTotalVoteCounts(
+  voteCounts: readonly ResultsReporting.VoteCounts[]
+): number {
+  return assertDefined(
+    voteCounts.find((vc) => vc.Type === CountItemType.Total),
+    'Could not find total vote count'
+  ).Count;
+}
+
+function convertToYesNoContest(
+  contest:
+    | ResultsReporting.BallotMeasureContest
+    // RetentionContest inherits from BallotMeasureContest.
+    // It also has additional fields for candidate that we ignore.
+    | ResultsReporting.RetentionContest
+): VxTabulation.YesNoContestResults {
+  const contestSelections = assertDefined(
+    contest.ContestSelection
+  ) as ResultsReporting.BallotMeasureSelection[];
+  const yesOption = findBallotMeasureSelectionWithContent(
+    /yes/i,
+    contestSelections
+  );
+  const noOption = findBallotMeasureSelectionWithContent(
+    /no/i,
+    contestSelections
+  );
+  const otherCounts = contest.OtherCounts && contest.OtherCounts[0];
+  const yesTally = findTotalVoteCounts(
+    assertDefined(
+      yesOption.VoteCounts,
+      'Could not find VoteCounts for "Yes" option'
+    )
+  );
+  const noTally = findTotalVoteCounts(
+    assertDefined(
+      noOption.VoteCounts,
+      'Could not find VoteCounts for "No" option'
+    )
+  );
+  const overvotes = otherCounts?.Overvotes || 0;
+  const undervotes = otherCounts?.Undervotes || 0;
+
+  return {
+    contestId: contest['@id'],
+    contestType: 'yesno',
+    yesOptionId: yesOption['@id'],
+    noOptionId: noOption['@id'],
+    yesTally,
+    noTally,
+    overvotes,
+    undervotes,
+    ballots: yesTally + noTally + overvotes + undervotes,
+  };
+}
+
+function getBallotCounts(contest: ResultsReporting.CandidateContest): {
+  overvotes: number;
+  undervotes: number;
+  total: number;
+} {
+  const otherCounts = contest.OtherCounts && contest.OtherCounts[0];
+
+  let ballotCount =
+    0 + (otherCounts?.Overvotes || 0) + (otherCounts?.Undervotes || 0);
```

Comment:
Are we sure that overvotes and undervotes represent individual ballots? They may represent a number of bubbles. E.g. undervotes may count each bubble that was not filled in. This is at least how we did things in our CVR CDF implementation. cc @adghayes

I don't see anything in the documentation about how to interpret this, but looking at our validation for manually entered tallies, it seems like we treat the overvotes/undervotes as vote counts, not ballot counts: https://github.com/votingworks/vxsuite/blob/main/apps/admin/frontend/src/screens/manual_data_entry_screen.tsx#L284-L304

I think the equivalency implied by that code is:
```
total ballots * votes allowed = candidate votes + overvotes + undervotes
```
Which would mean in this case, we should use this equation:
```
total ballots = (candidate votes + overvotes + undervotes) / votes allowed
```

Reply:
+1 to @jonahkagan's second formula

Reply:
https://github.com/votingworks/vxsuite/pull/5344/commits/56fdcc88c10c5ca012721f8db07f3f4a0fc42721